onnxruntime/test/common/tensor_op_test_utils.h

```diff
@@ -29,6 +29,17 @@ class RandomValueGenerator {
     return val;
   }
 
+  template <class T>
```

Comment:
a comment explaining the function above is for float [a, b) and this function is for int [a, b]

Reply:
perhaps we can use the signature: `vector<T> Uniform(const vector<int64_t>& dims, T min, T max)`
and dispatch to integral or floating point implementations with some template metaprogramming (e.g., with std::is_integral and std::is_floating_point).
if we keep them separate, naming them differently may be helpful. for example, UniformReal and UniformInt

Reply:
@edgchen1 I'm not sure that is a clean design because for floating types we will need to use `uniform_real_distribution` but for integer types we will need to use `uniform_int_distribution`. What do you think?

Reply:
yes, i was suggesting to keep a consistent signature, i.e., take T min and max values and return vector<T>, but use `uniform_real_distribution` or `uniform_int_distribution` in the implementation depending on T.

Reply:
i'm also fine with renaming this function to `UniformInt` or similar, or adding a comment as Sherlock suggested. otherwise it may be hard to easily see the difference between the two Uniform()'s

Reply:
Yea that seems a little messy to me. I had rather have it overloaded.

---
In reply to: [416953659](https://github.com/microsoft/onnxruntime/pull/3734#discussion_r416953659) [](ancestors = 416953659)

Reply:
I have added the comment.

---
In reply to: [416956029](https://github.com/microsoft/onnxruntime/pull/3734#discussion_r416956029) [](ancestors = 416956029)